Create microk8s clustering tutorial

rhys-the-davies opened the pull request:
A basic how-to for multi-node clustering in MicroK8s

## Done

[List of work items including drive-bys]

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8016/](http://0.0.0.0:8016/)
- [List additional steps to QA the new features or prove the bug has been resolved]


## Issue / Card

[List of links to Github issues/bugs and cards if needed - e.g. `Fixes #1`]

## Screenshots

[if relevant, include a screenshot]

rhys-the-davies committed:
Create microk8s clustering tutorial 

A basic how-to for multi-node clustering in MicroK8s

webteam-app commented:
User is not a collaborator of this repo. Please start demo manually.

webteam-app commented:
Starting demo at: https://tutorials-ubuntu-com-canonical-web-and-design-pr-1122.run.demo.haus/

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+tags: kubernetes, beginner, multipass, container, microk8s
+difficulty: 3
+status: draft
+published: 2019-09-13
```
This date needs to be more recent for the tutorial to be closer to the top of the page. Ideally matching more or less the release date of the tutorial (some time in jan 2020)

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+---
+id: raspberrypi-clustering-instructions
+summary: Create a raspberry pi cluser using microk8s
+categories: clusters
```
There is no "clusters" category, please refer to the tutorial guidelines, specifically the metadata section: https://tutorials.ubuntu.com/tutorial/tutorial-guidelines

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+---
+id: raspberrypi-clustering-instructions
```
This will end up being the URL of the tutorial, you may want to have a more descriptive URL for SEO purposes: eg. `rapsberry-pi-kubernetes-cluster`.

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+id: raspberrypi-clustering-instructions
+summary: Create a raspberry pi cluser using microk8s
+categories: clusters
+tags: kubernetes, beginner, multipass, container, microk8s
```
```suggestion
tags: kubernetes, multipass, container, microk8s, cluster
```

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+
+---
+
+Canonical is dedicated to enhancing the way we build and deliver technology through our continued investment in innovation and the open-source community. Over the past year, we’ve worked hard to realize the next phase of scaling and deployments on Kubernetes; we introduced [MicroK8s][microk8s], the lightweight, fast, powerful Kubernetes up in 60 seconds.
```
1) You need to start with a h1 level title, otherwise your tutorial won't have a title and won't appear as published.
2) This intro reads more like a PR than a tutorial. This is the first thing the audience will see, they may not know (or care) about Canonical, they just just know they want to create a Kubernetes cluster with their boards. Let's make sure the intro provides context based on their intent.

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+Canonical is dedicated to enhancing the way we build and deliver technology through our continued investment in innovation and the open-source community. Over the past year, we’ve worked hard to realize the next phase of scaling and deployments on Kubernetes; we introduced [MicroK8s][microk8s], the lightweight, fast, powerful Kubernetes up in 60 seconds.
+
+
+We’ve heard from our users, partners, and customers and today, we’re pleased and excited to announce multi-node clustering in MicroK8s. You can now join multiple deployments to form a cluster. Welcome to simple, enterprise-grade Kubernetes! 
```
Again, this isn't an announcement blog post.

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+
+We’ve heard from our users, partners, and customers and today, we’re pleased and excited to announce multi-node clustering in MicroK8s. You can now join multiple deployments to form a cluster. Welcome to simple, enterprise-grade Kubernetes! 
+
+# Test-Drive Time
```
Please review and follow the [tutorial guidelines](https://tutorials.ubuntu.com/tutorial/tutorial-guidelines). You need to define several things before jumping into this: requirements and objectives.

caldav commented on tutorials/containers/microk8s-clustering:
````diff
@@ -0,0 +1,71 @@
+microk8s.kubectl get no
+```
+
+And, you have successfully joined the node to the cluster! Repeat this microk8s.add-node and microk8s.join steps for each node you are adding.
````
```suggestion
You have successfully joined the node to the cluster! Repeat this microk8s.add-node and microk8s.join steps for each node you are adding.
```

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+# Test-Drive Time
+
+A preview release is available for you to test and give us your feedback. Lets set up the MicroK8s cluster.
+On two or more machines install MicroK8s from the 1.15/edge/clustering channel:
```
```suggestion
On two or more Raspberry Pis, install MicroK8s from the 1.15/edge/clustering channel:
```

caldav commented on tutorials/containers/microk8s-clustering:
```diff
@@ -0,0 +1,71 @@
+# Test-Drive Time
+
+A preview release is available for you to test and give us your feedback. Lets set up the MicroK8s cluster.
+On two or more machines install MicroK8s from the 1.15/edge/clustering channel:
```
What needs to be running on the PIs? What are the required setup steps before being able to run this command? cf. missing requirements.

rhys-the-davies replied:
This is a whole other tutorial I want to do as it's too many steps that aren't directly related to this tutorial. What's best, have brief instructions here, or wait and link?

caldav commented on tutorials/containers/microk8s-clustering:
````diff
@@ -0,0 +1,71 @@
+sudo microk8s.add-node
+```
+
+The microk8s.add-node will generate a connection string in the form of <master_ip>:<port>/<token> and will prompt you to use the microk8s.join command from the node you need to add to the cluster. 
````
You may want to add some network considerations to requirements: all Pis need to be on the same network, and you need to have terminal windows ready to connect to each through SSH.

rhys-the-davies replied:
Added this to the "what you'll need" section. Are there tutorials to link to that walk through connecting pis through SSH?

caldav commented on tutorials/containers/microk8s-clustering:
````diff
@@ -0,0 +1,71 @@
+You should be able to see the new node in a few seconds on the master. Run the following command:
+
+```bash
+microk8s.kubectl get no
````
`get no` or `get node`?

caldav commented on tutorials/containers/microk8s-clustering:
````diff
@@ -0,0 +1,71 @@
+sudo microk8s.remove <node-name>
+```
+
+Alternatively,  on the slave node you can run:
````
```suggestion
Alternatively, on the slave node you can run:
```

caldav commented on tutorials/containers/microk8s-clustering:
````diff
@@ -0,0 +1,71 @@
+sudo microk8s.leave
+```
+
+# What’s next
````
You can only have one h1 level title.
CF. tutorial guidelines to see which level refers to: #1 is main title, #2 new page title, #3 subheading in page.

caldav reviewed (changes requested):
Hey, there are many things to fix here, both from a structural and content point of view. Before addressing the comments, please read and follow the tutorial guidelines: https://tutorials.ubuntu.com/tutorial/tutorial-guidelines. Thanks!

Also, a screenshot of the final state (eg. output of the `status` command) would be appreciated. Bonus points for a nice picture of the physical PI cluster used to test the tutorial.

rhys-the-davies committed:
Update tutorials/containers/microk8s-clustering

Co-Authored-By: David Callé <david.calle@canonical.com>

rhys-the-davies committed:
Update tutorials/containers/microk8s-clustering

Co-Authored-By: David Callé <david.calle@canonical.com>

rhys-the-davies committed:
unfinished updates to the tutorial

Made most of the alterations caldav suggested. All of the ones regarding the tutorial for tutorials. Waiting for responses to the previous comments to complete the tutorial.

rhys-the-davies committed:
Update microk8s-clustering

steverydz commented:
Thanks for submitting a PR for this tutorial. We have moved our tutorials to discourse. To contribute going forwards please go to https://discourse.ubuntu.com/c/tutorials.
